lib/winapi/filemanip.c

```diff
@@ -235,3 +240,84 @@ BOOL MoveFileA (LPCTSTR lpExistingFileName, LPCTSTR lpNewFileName)
         return TRUE;
     }
 }
+
+DWORD GetLogicalDrives (VOID)
+{
+    DWORD result = 0;
+    NTSTATUS status;
+    ANSI_STRING path;
+    HANDLE handle;
+    OBJECT_ATTRIBUTES attributes;
+    struct {
+        OBJECT_DIRECTORY_INFORMATION objDirInfo;
+        CHAR filenameBuf[2];
+    } objDirInfoBuf;
+
+    InitializeObjectAttributes(&attributes, NULL, OBJ_CASE_INSENSITIVE, ObDosDevicesDirectory(), NULL);
+    status = NtOpenDirectoryObject(&handle, &attributes);
+
+    if (!NT_SUCCESS(status)) {
+        SetLastError(RtlNtStatusToDosError(status));
+        return 0;
+    }
+
+    ULONG index = 0;
+    while (true) {
+        POBJECT_DIRECTORY_INFORMATION objDirInfo = (POBJECT_DIRECTORY_INFORMATION)&objDirInfoBuf;
+        status = NtQueryDirectoryObject(handle, objDirInfo, sizeof(objDirInfoBuf), (index == 0), &index, NULL);
+
+        if (status == STATUS_NO_MORE_ENTRIES) {
+            break;
+        } else if (status == STATUS_BUFFER_TOO_SMALL) {
+            // If the object name is too long, we're not interested in it anyway
+            continue;
+        } else if (!NT_SUCCESS(status)) {
+            NtClose(handle);
+            SetLastError(RtlNtStatusToDosError(status));
+            return 0;
+        }
+
+        // Only allow entries like "C:\" or "E:\", ignore stuff like "CdRom0:"
```

Comment:
nit: Comment is wrong still, because it now has the backspace, which wouldn't fit into 2 characters which are being checked for. This might be my mistake (not sure what my review comment said).

Reply:
Oh, right. Fixed.